### DISCUSSION TITLE
Power Mask and DGT model integration in GNPy

### DISCUSSION THREAD

**Discussion by Khandujarnav** [Telecominfraproject/oopt-gnpy#510] [Q&A] [answered]
Please refer to the paper "A Comparison of DGT and Power Mask for EDFA Modeling in GNPy "(https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=10379688). 

The authors have updated GNPy library to consider the power mask approach of amplifiers in evaluations regarding optical links (ref: Conclusion section). 

The power mask is represented using a csv file where each line of the file contains the amplifier gain spectrum, the noise figure spectrum and the output power spectrum for a given total input power and target gain (ref: Simulation Setup section).

However, we could not find this code in the main branch. Kindly confirm if the authors have merged this code.

**Comment by AndreaDAmico** [comment, marked as answer]
Dear @Khandujarnav, 
the authors did not merge the code into GNPy repository. We would be happy to review the code in case the authors share their version. 
In any case, the inaccuracy of the DGT is a known issue (see "EDFA Gain Tilt Profile Error #471") and we are working to integrate a more accurate model described in the paper "Gain Profile Characterization and Modeling for Dual-Stage EDFA Abstraction and Control" (https://ieeexplore.ieee.org/abstract/document/10336837).
Indeed, we will compare the new implementation with the approach presented in "A Comparison of DGT and Power Mask for EDFA Modeling in GNPy ".

**Comment by tttlz** [comment]
Hello, I have read this thread and also #471. Is there any update on either fixing the issue or implementing the new model? Thank you
